src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(5);
+
+        switch (figureType) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
+                        random.nextDouble() * 10);

```

Comment:
why do you need to multiply each parameter

Reply:
I thought that random,nextDouble() returns value between 0 and 1.0, that's why I use multiplication in every case. Now I will try to rewrite code witout this.